### ISSUE TITLE
onnx's usage of INT64 instead of float seems to cause accuracy issues.

### ISSUE THREAD

**Issue by aifartist** [onnx/onnx-tensorrt#919]
## Description

Stable Diffusion models usually are fp32 or fp16.  So are TensorRT accelerated models.
To get from SD to TRT one goes through ONNX.  While most float weights appear to fall within the range from MIN_INT to MAX_INT, a few values can be quite large or be subnormal.  In the TRT convert process I see the following messages:
```
[W] onnx2trt_utils.cpp:374: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[W] onnx2trt_utils.cpp:400: One or more weights outside the range of INT32 was clamped

[W] TensorRT encountered issues when converting weights between types and that could affect accuracy.
[W] - 110 weights are affected by this issue: Detected subnormal FP16 values.

```
Clamping a huge value to MAX_INT32 is a problem accuracy.  If you used ONNX float16 you wouldn't lose the magnitude of very large values.  Yes, the precision is less but given the model eventually is output in fp16 form why would this matter?
The bottom line is that Stable Diffusion fp16 models produce very crisp images but something in the act of converting from fp16/fp32 -> INT64 -> INT32 -> fp16(TRT) is producing poor quality images.

I can't upload a 4GB Stable Diffusion model as a repro.  However, the math is obvious and clearly accuracy is lost if you go from float to int to float.  Given that ONNX supports fp16 or fp32 why does torch.onnx.export not have an option to keep the data in float?

This is just basic math.
ONNX supports float.
But torch.onnx.export() doesn't expose that as an option.
A repro would be huge.
If it really matters I have a 4090 and TRT 8.6.1 and pytorch 1.13.1 on Ubuntu 22.04 but that doesn't change the math.
